virttest/utils_net.py

```diff
@@ -1697,20 +1697,31 @@ def restart_guest_network(
     :param timeout: timeout value for command.
     """
     if os_type == "linux":
+        dhcp_clients = [("dhclient", "-r"), ("dhcpcd", "-k")]
+        for client, release_arg in dhcp_clients:
+            try:
+                cmd = "which %s" % client
+                dhcp_cmd_path = session.cmd(cmd).strip()
+                if dhcp_cmd_path:
+                    dhcp_cmd = client
+                    release_cmd = "%s %s" % (dhcp_cmd, release_arg)
+                    break
+            except aexpect.ShellCmdError:
+                continue
```

Comment:
```suggestion
            try:
                session.cmd("which %s" % client)
            except aexpect.ShellCmdError:
                continue
            release_cmd = "%s %s" % (client, release_arg)
            break
```
I think that we can simplify the code with the above.
For your implementation, the statement `dhcp_cmd_path = session.cmd(cmd).strip()`， I think it is a bit redundant.
Maybe you have your thoughts about it, please let me know if I misunderstood here.

Reply:
When the system has only the dhcpcd command but without dhclient,  in the first loop, the release_cmd = "dhclient -r", and the it will break. it's not expected.

Reply:
> In your suggestion code, when the system has only the dhcpcd command but without dhclient, in the first loop, the release_cmd = "dhclient -r", and the it will break. it's not expected.

It will raise an exception `ShellCmdError` if the system without `dhclient` since the return code is `1` of the command `which dhclient` and the `session.cmd` will raise the `ShellCmdError` if the exit status is nonzero.

```
# which dhclient
/usr/sbin/dhclient
# echo $?
0
# which dhcpcd
/usr/bin/which: no dhcpcd in (/home/yhong/kar/workspace/bin:/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin)
# echo $
1
```

Reply:
ok, I have updated, please help review. 

I also test with rhel9 and latest vm, it's passed. 

rhel9:
[stdlog] 2024-08-04 23:17:40,720 aexpect.client client           L1184 DEBUG| Sending command: which dhclient
[stdlog] 2024-08-04 23:17:40,826 aexpect.client client           L1184 DEBUG| Sending command: echo $?
[stdlog] 2024-08-04 23:17:40,927 aexpect.client client           L1223 DEBUG| Sending command (safe): dhclient -r; dhclient

latest vm:
[stdlog] 2024-08-04 23:12:28,382 aexpect.client client           L1184 DEBUG| Sending command: which dhclient
[stdlog] 2024-08-04 23:12:28,487 aexpect.client client           L1184 DEBUG| Sending command: echo $?
[stdlog] 2024-08-04 23:12:28,588 aexpect.client client           L1184 DEBUG| Sending command: which dhcpcd
[stdlog] 2024-08-04 23:12:28,692 aexpect.client client           L1184 DEBUG| Sending command: echo $?
[stdlog] 2024-08-04 23:12:28,793 aexpect.client client           L1223 DEBUG| Sending command (safe): dhcpcd -k; dhcpcd